packages/walletconnect-solana/src/adapter.ts

```diff
@@ -115,16 +117,52 @@ export class WalletConnectWallet {
         }
     }
 
-    async signTransaction(transaction: Transaction): Promise<Transaction> {
+    async signTransaction<T extends Transaction | VersionedTransaction>(transaction: T): Promise<T> {
         if (this._client && this._session) {
-            const { signature } = await this._client.request<{ signature: string }>({
-                chainId: this._network,
-                topic: this._session.topic,
-                request: { method: WalletConnectRPCMethods.signTransaction, params: { ...transaction } },
-            });
-            transaction.addSignature(this.publicKey, Buffer.from(base58.decode(signature)));
-
-            return transaction;
+            if (isVersionedTransaction(transaction)) {
+                const rawTransaction = Buffer.from(transaction.serialize()).toString('base64');
+                if (transaction.version === 'legacy') {
+                    const legacyTransaction = Transaction.from(transaction.serialize());
+                    const { signature } = await this._client.request<{ signature: string }>({
+                        chainId: this._network,
+                        topic: this._session.topic,
+                        request: {
+                            method: WalletConnectRPCMethods.signTransaction,
+                            params: { ...legacyTransaction, transaction: rawTransaction },
+                        },
+                    });
+                    transaction.addSignature(this.publicKey, base58.decode(signature));
+                    return transaction;
+                } else {
+                    const { signature } = await this._client.request<{ signature: string }>({
+                        chainId: this._network,
+                        topic: this._session.topic,
+                        request: {
+                            method: WalletConnectRPCMethods.signTransaction,
+                            params: { transaction: rawTransaction },
+                        },
+                    });
+                    transaction.addSignature(this.publicKey, base58.decode(signature));
+                    return transaction;
+                }
+            } else {
+                const rawTransaction = transaction
+                    .serialize({
+                        requireAllSignatures: false,
+                        verifySignatures: false,
+                    })
+                    .toString('base64');
+                const { signature } = await this._client.request<{ signature: string }>({
+                    chainId: this._network,
+                    topic: this._session.topic,
+                    request: {
+                        method: WalletConnectRPCMethods.signTransaction,
+                        params: { ...transaction, transaction: rawTransaction },
```

Comment:
the previous version did not require an explicit serialization of the transaction, is there a particular reason why that's needed for the existing non-versioned transactions?

Reply:
@jnwng Versioned Transaction and legacy Transaction have completely different set of properties. Passing non-serialized transaction will require to create different methods for the future changes of Transaction class. 

Also it makes its much harder to consume deconstructed transactions on the wallet side.  

To make API future proof and easy to use we believe that moving to serialized transaction is a better approach, which is similar to approach of Solana Pay.